modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -717,31 +717,34 @@ def begin_date_exception(idx)
         )
       end
 
-      def detect_invalid_active_duty_enddate(service_information)
-        service_information['servicePeriods'].detect do |service_period|
-          errors = date_is_valid?(service_period['activeDutyEndDate'],
-                                  'serviceInformation/servicePeriods/activeDutyEndDate', true)
-          return true if errors.is_a?(Array)
+      def validate_form_526_location_codes(service_information)
+        service_periods = service_information['servicePeriods']
+        any_code_present = service_periods.any? do |service_period|
+          service_period['separationLocationCode'].present?
         end
-      end
 
-      def validate_form_526_location_codes(service_information)
         # only retrieve separation locations if we'll need them
-        invalid_end_date = detect_invalid_active_duty_enddate(service_information).is_a?(Array)
+        return unless any_code_present
 
-        need_locations = service_information['servicePeriods'].any? do |service_period|
-          service_period['separationLocationCode'].present?
+        separation_locations = retrieve_separation_locations
+
+        if separation_locations.nil?
+          collect_error_messages(
+            detail: 'The Reference Data Service is unavailable to verify the separation location code for the claimant'
+          )
+
+          return
```

Comment:
You will need to get an index in this error message in case there are multiple examples (not sure how much that happens but it is possible). We want the error message to include the index of the incorrect submission(s) and should also point to the specific value in the form.

An example would be something similar
```
collect_error_messages(source: "/disabilities/#{idx}/name",
               detail: "The disability name is required for /disabilities/#{idx}/name")
```

So I guess something similar to
```
source: "/serviceInformation/servicePeriods#{idx}/seperarationLocationCode",
detail: " seperarationLocationCode for .... with index ... rest of the message you have in there"
```
^ pseudo code obviously but kind of the idea of what we would want for that.